### PR TITLE
Fix interop `FieldGet` & `FieldSet` methods with default parameters

### LINKED ISSUE #36724: `ArrayIndexOutOfBoundsException` when using default value parameters with function call in interop FieldSet
**Description:**

$title
```
[2022-06-24 11:23:53,032] SEVERE {b7a.log.crash} - Index -1 out of bounds for length 20 
java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 20
        at org.objectweb.asm.Frame.getLocal(Frame.java:486)
        at org.objectweb.asm.Frame.execute(Frame.java:768)
        at org.objectweb.asm.MethodWriter.visitVarInsn(MethodWriter.java:926)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmCodeGenUtil.genYieldCheck(JvmCodeGenUtil.java:621)
        at org.wso2.ballerinalang.compiler.bir.codegen.interop.InteropMethodGen.generateBasicBlocks(InteropMethodGen.java:313)
        at org.wso2.ballerinalang.compiler.bir.codegen.interop.InteropMethodGen.genJFieldForInteropField(InteropMethodGen.java:192)
        at org.wso2.ballerinalang.compiler.bir.codegen.interop.ExternalMethodGen.genJMethodForBExternalFunc(ExternalMethodGen.java:75)
        at org.wso2.ballerinalang.compiler.bir.codegen.methodgen.MethodGen.generateMethod(MethodGen.java:159)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.lambda$generateModuleClasses$0(JvmPackageGen.java:446)
        at java.base/java.util.HashMap$EntrySet.forEach(HashMap.java:1039)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generateModuleClasses(JvmPackageGen.java:397)
        at org.wso2.ballerinalang.compiler.bir.codegen.JvmPackageGen.generate(JvmPackageGen.java:808)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:95)
        at org.wso2.ballerinalang.compiler.bir.codegen.CodeGenerator.generate(CodeGenerator.java:70)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:309)
        at io.ballerina.projects.ModuleContext.generateCodeInternal(ModuleContext.java:473)
        at io.ballerina.projects.ModuleCompilationState$4.generatePlatformSpecificCode(ModuleCompilationState.java:132)
        at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:381)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:165)
        at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:134)
        at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:112)
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
        at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:169)
        at io.ballerina.projects.JBallerinaBackend.from(JBallerinaBackend.java:111)
        at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:122)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.RunCommand.execute(RunCommand.java:205)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:51)
```

**Steps to reproduce:**
java code (in "jar_dependencies/testBalFieldSet.jar")
```java
public class TestFieldSet{

    public static long a = 12;

    public static long geta() {
        return a;
    }

}
```

ballerina code
```bal
import ballerina/jballerina.java;
import ballerina/io;

public function get_a() returns int = @java:FieldGet {
   name:"a",
   'class:"TestFieldSet"
} external;

function setA(int a = (1+2+func())) = @java:FieldSet {
  name:"a",
  'class:"TestFieldSet"
} external;

function func() returns int {
    return 3;
}

public function main() {
   setA();
   int l = get_a();
   io:println(l);
}
```
Ballerina.toml
```
[[platform.java11.dependency]]
path = "jar_dependencies/testBalFieldSet.jar"
```

**Affected Versions:**
Ballerina 2201.1.0 (Swan Lake)

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #36724

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/javainterop/basic/field_access_mutate_tests.bal [hunk 1]
```diff
@@ -1,4 +1,21 @@
+// Copyright (c) 2019, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
```
Comment (warunalakshitha) [inline tests/jballerina-unit-test/src/test/resources/test-src/javainterop/basic/field_access_mutate_tests.bal:1]:
```suggestion
// Copyright (c) 2019-2022, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
```

Comment (Nadeeshan96) [reply tests/jballerina-unit-test/src/test/resources/test-src/javainterop/basic/field_access_mutate_tests.bal:1]: Kept as 2019 as discussed.

**Review by HindujaB — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**